### PR TITLE
chore(deps): update ruff to 0.1.1 and fix lint issues

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `make lint`?
- [x] Have you successfully run `pytest tests/unit`?

-----

### CODE REVIEW

File: snapcraft/parts/grammar.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def process_parts(
     processor = GrammarProcessor(
         arch=arch,
         target_arch=target_arch,
-        checker=lambda x: x == x,  # pylint: disable=comparison-with-itself
+        checker=lambda x: x == x,  # pylint: disable=comparison-with-itself  # noqa PLR0124
```
Comment (mr-cal) [inline snapcraft/parts/grammar.py:71]: Looks like black is unhappy with this line

Comment (lengau) [reply snapcraft/parts/grammar.py:71]: "Fixing one linter broke another linter"

A software engineering play in 2 acts

Comment (lengau) [reply snapcraft/parts/grammar.py]: OK, linter collisions fixed I think.

File: snapcraft/parts/grammar.py [hunk 2]
```diff
@@ -64,12 +64,12 @@ def process_parts(
     :param yaml_data: unprocessed snapcraft.yaml.
     :returns: process snapcraft.yaml.
     """
+
+    def self_check(x: Any) -> bool:
+        return x == x  # pylint: disable=comparison-with-itself  # noqa PLR0124
```
Comment (syu-w) [inline snapcraft/parts/grammar.py:69]: pylint: C0103: Argument name "x" doesn't conform to snake_case naming style (invalid-name)

File: pyproject.toml [hunk 3]
```diff
@@ -77,6 +77,7 @@ addopts = ["--cov=snapcraft"]
 [tool.ruff]
 target-version = "py310"
 line-length = 88
+target-version = "py310"
```
Comment (mr-cal) [inline pyproject.toml:80]: Isn't this already here 2 lines above?

Comment (mr-cal) [reply pyproject.toml:80]: Strangely enough, it looks like this is causing the errors in CI of `KeyError: 'lint-docs'`

Comment (lengau) [reply pyproject.toml:80]: This is what I get for rebasing without doing a thorough check of the results!

```suggestion
```

File: snapcraft/parts/grammar.py (hunk 2 above)
Comment (lengau) [reply snapcraft/parts/grammar.py]: Renamed to `value`. I wonder why I didn't get this when I linted locally 🤔

**Comment by syu-w** [comment]
ruff seems just got 0.1.2 😂

**Comment by lengau** [comment]
@syu-w well fortunately that update should be far easier!

**Review by syu-w — APPROVED** [review]

**Pull request merged**